src/jit/assertionprop.cpp

```diff
@@ -957,7 +957,7 @@ AssertionIndex Compiler::optCreateAssertion(GenTree*         op1,
                 }
             }
 
-            if (fgIsBigOffset(offset) || (vnStore->TypeOfVN(vn) != TYP_REF))
```

Comment:
You also should update the comment on line: 926:

`	        // so only make non-null assertions about GC refs`

Reply:
Done.